pulsar-client/src/main/java/org/apache/pulsar/client/api/AuthenticationFactory.java

```diff
@@ -81,8 +65,9 @@ public static final Authentication create(String authPluginClassName, String aut
      * @param authParams          map which represents parameters for the Authentication-Plugin
      * @return instance of the Authentication-Plugin
      * @throws UnsupportedAuthenticationException
+     * @deprecated Use create(String authPluginClassName, String authParamsString)
      */
-    @SuppressWarnings("deprecation")
+    @Deprecated
     public static final Authentication create(String authPluginClassName, Map<String, String> authParams)
```

Comment:
@maskit While I agree we don't need 2 ways to configure the authentication for plugins, could we leave it for users? We can still encode the map internally and pass it to the plugin. 

In some cases, if you're not getting the auth settings from a file, but rather set in code, for a user it's easier to fill up a map than then figure out the specific encoding

Reply:
Hmm, yeah, we should consider the use case. Let me think about that. Passing a ‘Map’ doesn’t sound correct to me. Maybe it should be ‘Object’.

For now, I’ll update this PR within a few
 days to leave the method.